nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/PutEmail.java

```diff
@@ -319,6 +330,15 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
             message.setRecipients(RecipientType.CC, toInetAddresses(context, flowFile, CC));
             message.setRecipients(RecipientType.BCC, toInetAddresses(context, flowFile, BCC));
 
+            final String attributeNameRegex = context.getProperty(ATTRIBUTE_NAME_REGEX).getValue();
+            final Pattern attributeNamePattern = attributeNameRegex == null ? null : Pattern.compile(attributeNameRegex);
+            if (attributeNamePattern != null) {
+                for (final Map.Entry<String, String> entry : flowFile.getAttributes().entrySet()) {
+                    if (attributeNamePattern.matcher(entry.getKey()).matches()) {
```

Comment:
There are rules about how the headers have to be encoded.  We should use the MimeUtility to ensure everything is encoded correctly.

```
Note that RFC 822 headers must contain only US-ASCII characters, so a header that contains non US-ASCII characters must have been encoded by the caller as per the rules of RFC 2047.
```

[MimeUtility](https://docs.oracle.com/javaee/6/api/javax/mail/internet/MimeUtility.html)

We should also have tests for ensuring attributes with content that must be encoded are handled.